### ISSUE TITLE
CMake Error with XSD_LIBRARY when manually compiling

### ISSUE THREAD

**Issue by chuchro3** [Microsoft/malmo#421]
Hi, I'm experiencing a similar issue to #208 , but unlike the reporter, I actually have xsd installed on my system and am still receiving the following output:

```
$ xsd version
CodeSynthesis XSD XML Schema to C++ compiler 4.0.0
Copyright (c) 2005-2014 Code Synthesis Tools CC
```

```
$ cmake -DCMAKE_BUILD_TYPE=Debug ..
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   chrono
--   date_time
--   filesystem
--   iostreams
--   program_options
--   python
--   regex
--   system
--   thread
--   atomic
-- Found the following C# Mono versions: 4.6.2
-- Found CSharp: Mono  
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") 
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- Found Java: /usr/lib/jvm/java-8-openjdk-amd64/bin/java (found version "1.8.0.111") found components:  Runtime Development 
-- Found JNI: /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/libjawt.so  
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.12", minimum required is "2.7") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found suitable version "2.7.12", minimum required is "2.7") 
-- Found SWIG: /usr/bin/swig3.0 (found version "3.0.8") 
-- Found Luabind: /usr/local/lib/libluabind.a  
-- Found Torch7 in /home/robert/torch/install
-- Using Mono compiler version 4.6.2
-- Found the following C# Mono versions: 4.6.2
-- Using Mono compiler version 4.6.2
-- Found the following C# Mono versions: 4.6.2
-- Using Mono compiler version 4.6.2
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
XSD_LIBRARY (ADVANCED)
    linked by target "Malmo" in directory /home/robert/repos/MalmoPlatform/Malmo/src
```

Using Ubuntu 16.04

Any advice to push me in the right direction would be great, I feel stuck at this point.

**Comment by DaveyBiggers** [comment]
Can I ask how you installed CodeSynthesis? I realise our build instructions don't have a special section for Ubuntu 16.04 (https://github.com/Microsoft/malmo/blob/master/doc/build_linux.md) - so did you follow the instructions for Ubuntu 15 (and install using `sudo apt-get install xsdcxx `), or did you follow the Ubuntu 14 instructions (with the `wget` and `sudo dpkg -i --force-all xsd_4.0.0-1_amd64.deb`)?

The extra instructions for Ubuntu 14/Debian 7 are needed because the packaged version of xsd is too old - we require version 4.0. Installing version 4 this way results in the binary being called "xsd". On Ubuntu 15 and 16, Debian 8 etc. version 4.0 comes as standard, and is called "xsdcxx". CMake decides which to look for ("xsd" or "xsdcxx") by checking which OS it's running on.

So my best guess at what's happening is that CMake is looking for `xsdcxx`, whereas your version is installed as `xsd`. (To confirm this suspicion, if I type `xsd version` in my copy of Ubuntu 16, I get `Invalid parameters` - this is the Mono xsd. On the other hand, typing `xsdcxx version` gives me what you posted above.)

If this is so, a simple `sudo apt-get install xsdcxx` _should_ fix things.

**Comment by chuchro3** [comment]
I removed xsd because attempting to install xsdcxx complained about directory conflicts, then installed xsdcxx successfully.

I had to backtrack and reinstall a couple other packages, and it finally compiled!

@DaveyBiggers It looks like this happened because I was originally following the tutorial to use the precompiled version, which doesn't have you install xsdcxx, and when I switched over to trying to build it manually, it created some confusion since I already had (the wrong) xsd on my system.

Thanks for the help, now it's time to have some fun!
